### DISCUSSION TITLE
context.Database | FromSqlRaw methods are not accessible when using this template.

### DISCUSSION THREAD

**Discussion by yaseerngr** [jasontaylordev/CleanArchitecture#451] [Q&A] [answered]
Hi @jasontaylordev ,

Thanks for this wondering template, I'm truly one of those who are benefitted by using this template.

I'm having question regarding running raw sql in your template. I'm not able to access FromSqlRaw from context

I'm trying to run a separate sql statement like below

`_context.TodoItems
                .FromSqlRaw("SELECT * FROM dbo.TodoItems")
                .ToList();`

But .FromSqlRaw is not accessible. 

context.database is also not accessible. Any idea what is stopping me to access these methods in this template.

**Comment by fretje** [comment]
That is because _context here is an `IApplicationDbContext` which doesn't expose things like `Database` or `FromSqlRaw`. 

You could add those to the interface, but that would not be a clean solution.

If you need that kind of specific behaviour (FromRawSql), you should implement that in the infrastructure layer, as otherwise you're making your application layer dependent on a specific database engine. Create an interface for it in the application layer which you then use there.

**Reply by yaseerngr** [reply]
Thanks for the reply. Now I can understand. I'm going to add that as a separate method in the ApplicationDbcontext as suggested in this [#248 (comment)](https://github.com/jasontaylordev/CleanArchitecture/issues/248#issuecomment-733670024)

**Comment by artworkad** [comment, marked as answer]
https://github.com/jasontaylordev/CleanArchitecture/issues/248#issuecomment-733670024

**Comment by suicodave** [comment]
I don't know why that happens to you but to me, it works just perfectly fine. Take a look at this screenshot
![fromsqlraw](https://user-images.githubusercontent.com/28585323/156275591-886856b1-f0a8-4d65-8351-a2ad9ef19e34.png)

**Reply by fretje** [reply]
That's because you probably referenced EntityFrameworkCore into your application project, which you shouldn't do, as the application project shouldn't be dependent on any infrastructure stuff (which EntityFrameworkCore is).

**Reply by suicodave** [reply]
Oh I see, you got it right there
